### ISSUE TITLE
Questions about how to program the COP421

### ISSUE THREAD

**Issue by Tachibana-Camellia** [rulecity/COP421Reader#1]
@rulecity Thank you for your sharing, but I’ve encountered a small problem. After obtaining the .bin file of the internal program for the COP421, I’m not sure how to write it into a new COP421 chip. I’ve consulted the TI COP400 series data manual from that era, but it doesn’t mention this at all. I’m at a loss and have no choice but to seek your help. If you’re willing to provide me with some guidance, I would be extremely grateful. I apologize again for the inconvenience.
<img width="190" alt="4" src="https://github.com/user-attachments/assets/03be6b97-2e17-4924-80a9-7ba7a66930ba">

**Comment by Tachibana-Camellia** [comment]
I thought I 've made a mistake. COP421 has a mask ROM which can't be programmed.

**Comment by rulecity** [comment]
Yes, the COP421 can only be programmed one time, I assume at the factory.

Out of curiosity, how did you obtain the .bin file?  I assume you did not use my project, since some of the source code for the desktop application is missing.

**Comment by Tachibana-Camellia** [comment]
One of my friends is attempting to replicate the Apple Lisa 2. He provided me with the assembly source code from the COP421 ROM, and the organizer of that source code is listed as Dr. Patrick Schäfer. So, I suspect he obtained the source code from Dr. Patrick's blog.
